aten/src/ATen/native/SummaryOps.cpp

```diff
@@ -52,10 +52,79 @@ Tensor _bincount_cpu_template(
   }
   return output;
 }
+
+// A helper function to get the bin for histogram
+// while each bin is inclusive at the lower end and exclusive at the higher,
+// i.e. [start, end) the last bin is inclusive at both, i.e. [start, end], in
+// order to include maxvalue if exists
+template <typename input_t>
+inline int64_t getbin(input_t x, input_t min, input_t max, int64_t bins) {
+  if (x == max)
+    return bins - 1;
+  return static_cast<int64_t>((x - min) * bins / (max - min));
+}
+
+///////////////// histogram /////////////////
+template <typename input_t, typename weights_t>
+Tensor _histogram_cpu_template_uniformbins(
+    const Tensor& self,
+    int64_t nbins,
+    const Tensor& weights,
+    input_t min,
+    input_t max,
+    bool density) {
+  // This is done to avoid divide by zero. Technically, it results in incorrect behavior in that case,
+  // but it is consistent with numpy.
+  if (min == max) {
+    min -= .5;
+    max += .5;
+  }
+  TORCH_CHECK(nbins > 0, "bins must be > 0");
+  TORCH_CHECK(
+      !(std::isinf(static_cast<double>(min)) ||
+        std::isinf(static_cast<double>(max)) ||
+        _isnan(min) || _isnan(max)),
+      "range of [",
+      min,
+      ", ",
+      max,
+      "] is not finite");
+  TORCH_CHECK(min < max, "max must be larger than min");
+
+  bool has_weights = weights.defined();
+  if (has_weights && weights.size(0) != self.size(0)) {
+    AT_ERROR("input and weights should have the same length");
+  }
+  Tensor output;
+  int64_t self_size = self.size(0);
+
+  const input_t* self_p = self.data_ptr<input_t>();
```

Comment:
How does this work if self is not contiguous?

Reply:
This assumes self is contiguous, added a comment.